packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -325,12 +325,30 @@ export class DeltaManager
      * as in read-only permissions.
      * But this.active can be used by some DDSes to figure out if ops can be sent
      * (for example, read-only view still participates in code proposals / upgrades decisions)
+     *
+     * Forcing Readonly does not prevent DDS from generating ops. It is up to user code to honour
+     * the readonly flag. If ops are generated, they will accumulate locally and not be sent. If
+     * there are pending in the outbound queue, it will stop sending until force readonly is
+     * cleared.
+     *
+     * @param readonly - set or clear force readonly.
      */
     public forceReadonly(readonly: boolean) {
         const oldValue = this.readonly;
         this._forceReadonly = readonly;
         if (oldValue !== this.readonly) {
+            let reconnect = false;
```

Comment:
I'm not certain reconnection is right flow.
We might be already connected as "read"
Also re-connecting as "read" can still give you "write" connection - server is free to give you more.

Also this._reconnectMode might be ReconnectMode.Never or ReconnectMode.Disabled, in such case we should not reconnect (this is checked in reconnectOnError(), which probably should be duplicated via assert connectCore(), but it also likely point out that we are better off using reconnectOnError() flow vs. manually disconnecting and reconnecting.

I'd rather either scope the usage of this API to disconnected state, OR not touch connection, but operate only with readonly property. Latter is simpler and feels better, but it breaks the assumption that we somewhat ride on here - that connection properties (like permissions, read-only, etc.) are changed only on granularity of connections.

Reply:
I would only reconnect if we switch from write to read, and it really does matter what permission the server give us.  We don't rely on that to stop sending ops to the server, but explicity depend on the readonly flag.

Not touching the connection is where we have problems with the assert in the readonly event handler, because we can't replay the ops after we reconnect on the same connection.

Logically, this is an internal disconnect reconnect, not auto reconnect which I thought the reconnectMode is supposed to control.  What is the usage of reconnectMode?

Reply:
It makes sense. One thing to consider though - this is not fully invisible to users. For example, if that functionality would have been applied to summarizing container, then it would puke, because it notices disconnects and stops.
But I guess we can leave with it.

Reply:
True..... may be we can disable this API for summarizing clients?   I will leave it for now.